R/markdown.R

```diff
@@ -72,11 +72,20 @@ markdown_text <- function(text, pkg = pkg, ...) {
 
 
 markdown_text2 <- function(text, pkg, ...) {
+
+  if (is.null(text)) {
+    return(NULL)
+  }
+
   html <- markdown_text(text, pkg = pkg, ...)
-  html %>%
+  children <- html %>%
     xml2::read_html() %>%
     xml2::xml_child() %>% # body
-    xml2::xml_children() %>% # p
-    as.character() %>%
-    paste(collapse = "")
+    xml2::xml_children()
+
+  if (length(children) == 1) {
```

Comment:
Let's leave as in this PR, but I'm not convinced this is the correct logic, so can you please do another PR that adds a few tests?

Reply:
https://github.com/r-lib/pkgdown/issues/1529